In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import warnings
warnings.filterwarnings('ignore')
pd.set_option('display.max_rows',None)
pd.set_option('display.max_columns',None)

In [2]:
data = pd.read_csv('Data file.csv')
data.head()

Unnamed: 0  label       msisdn    aon  daily_decr30  daily_decr90  \
0           1      0  21408I70789  272.0   3055.050000   3065.150000   
1           2      1  76462I70374  712.0  12122.000000  12124.750000   
2           3      1  17943I70372  535.0   1398.000000   1398.000000   
3           4      1  55773I70781  241.0     21.228000     21.228000   
4           5      1  03813I82730  947.0    150.619333    150.619333   

   rental30  rental90  last_rech_date_ma  last_rech_date_da  last_rech_amt_ma  \
0    220.13    260.13                2.0                0.0              1539   
1   3691.26   3691.26               20.0                0.0              5787   
2    900.13    900.13                3.0                0.0              1539   
3    159.42    159.42               41.0                0.0               947   
4   1098.90   1098.90                4.0                0.0              2309   

   cnt_ma_rech30  fr_ma_rech30  sumamnt_ma_rech30  medianamnt_ma_rech30  \
0              2          21.0             3078.0                1539.0   
1              1           0.0             5787.0                5787.0   
2              1           0.0             1539.0                1539.0   
3              0           0.0                0.0                   0.0   
4              7           2.0            20029.0                2309.0   

   medianmarechprebal30  cnt_ma_rech90  fr_ma_rech90  sumamnt_ma_rech90  \
0                  7.50              2            21               3078   
1                 61.04              1             0               5787   
2                 66.32              1             0               1539   
3                  0.00              1             0                947   
4                 29.00              8             2              23496   

   medianamnt_ma_rech90  medianmarechprebal90  cnt_da_rech30  fr_da_rech30  \
0                1539.0                  7.50            0.0           0.0   
1                5787.0                 61.04            0.0           0.0   
2                1539.0                 66.32            0.0           0.0   
3                 947.0                  2.50            0.0           0.0   
4                2888.0                 35.00            0.0           0.0   

   cnt_da_rech90  fr_da_rech90  cnt_loans30  amnt_loans30  maxamnt_loans30  \
0              0             0            2            12              6.0   
1              0             0            1            12             12.0   
2              0             0            1             6              6.0   
3              0             0            2            12              6.0   
4              0             0            7            42              6.0   

   medianamnt_loans30  cnt_loans90  amnt_loans90  maxamnt_loans90  \
0                 0.0          2.0            12                6   
1                 0.0          1.0            12               12   
2                 0.0          1.0             6                6   
3                 0.0          2.0            12                6   
4                 0.0          7.0            42                6   

   medianamnt_loans90  payback30  payback90 pcircle       pdate  
0                 0.0  29.000000  29.000000     UPW  2016-07-20  
1                 0.0   0.000000   0.000000     UPW  2016-08-10  
2                 0.0   0.000000   0.000000     UPW  2016-08-19  
3                 0.0   0.000000   0.000000     UPW  2016-06-06  
4                 0.0   2.333333   2.333333     UPW  2016-06-22

In [3]:
data.shape

(209593, 37)

we are having 209593 observations in or data set and 36 independent variables 1 dependent variable (label)

### DATA INFO

In [4]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 209593 entries, 0 to 209592
Data columns (total 37 columns):
 #   Column                Non-Null Count   Dtype  
---  ------                --------------   -----  
 0   Unnamed: 0            209593 non-null  int64  
 1   label                 209593 non-null  int64  
 2   msisdn                209593 non-null  object 
 3   aon                   209593 non-null  float64
 4   daily_decr30          209593 non-null  float64
 5   daily_decr90          209593 non-null  float64
 6   rental30              209593 non-null  float64
 7   rental90              209593 non-null  float64
 8   last_rech_date_ma     209593 non-null  float64
 9   last_rech_date_da     209593 non-null  float64
 10  last_rech_amt_ma      209593 non-null  int64  
 11  cnt_ma_rech30         209593 non-null  int64  
 12  fr_ma_rech30          209593 non-null  float64
 13  sumamnt_ma_rech30     209593 non-null  float64
 14  medianamnt_ma_rech30  209593 non-null  float64
 15  

most of our variables are numeric, pdate variable needs to be changed to datetime format to extract useful values.

Unnamed: 0, pcircle, and  msisdn are not necessary for our prediction 
pcircle = it has only one value in all the observation, there is no variance so removing this will be a better approach.
Unnamed: 0 and msisdn are there only to locate the indivisual observation.

In [5]:
data = data.drop(['Unnamed: 0','msisdn','pcircle'],axis=1)
#Extracting date and month from pdate column
data['pdate'] = pd.to_datetime(data['pdate'])
data['date'] = data['pdate'].dt.day
data['month'] = data['pdate'].dt.month
data.drop('pdate',axis=1,inplace=True)

### Statistical description

In [6]:
data.describe()

label            aon   daily_decr30   daily_decr90  \
count  209593.000000  209593.000000  209593.000000  209593.000000   
mean        0.875177    8112.343445    5381.402289    6082.515068   
std         0.330519   75696.082531    9220.623400   10918.812767   
min         0.000000     -48.000000     -93.012667     -93.012667   
25%         1.000000     246.000000      42.440000      42.692000   
50%         1.000000     527.000000    1469.175667    1500.000000   
75%         1.000000     982.000000    7244.000000    7802.790000   
max         1.000000  999860.755168  265926.000000  320630.000000   

            rental30       rental90  last_rech_date_ma  last_rech_date_da  \
count  209593.000000  209593.000000      209593.000000      209593.000000   
mean     2692.581910    3483.406534        3755.847800        3712.202921   
std      4308.586781    5770.461279       53905.892230       53374.833430   
min    -23737.140000  -24720.580000         -29.000000         -29.000000   
25%       280.420000     300.260000           1.000000           0.000000   
50%      1083.570000    1334.000000           3.000000           0.000000   
75%      3356.940000    4201.790000           7.000000           0.000000   
max    198926.110000  200148.110000      998650.377733      999171.809410   

       last_rech_amt_ma  cnt_ma_rech30   fr_ma_rech30  sumamnt_ma_rech30  \
count     209593.000000  209593.000000  209593.000000      209593.000000   
mean        2064.452797       3.978057    3737.355121        7704.501157   
std         2370.786034       4.256090   53643.625172       10139.621714   
min            0.000000       0.000000       0.000000           0.000000   
25%          770.000000       1.000000       0.000000        1540.000000   
50%         1539.000000       3.000000       2.000000        4628.000000   
75%         2309.000000       5.000000       6.000000       10010.000000   
max        55000.000000     203.000000  999606.368132      810096.000000   

       medianamnt_ma_rech30  medianmarechprebal30  cnt_ma_rech90  \
count         209593.000000         209593.000000   209593.00000   
mean            1812.817952           3851.927942        6.31543   
std             2070.864620          54006.374433        7.19347   
min                0.000000           -200.000000        0.00000   
25%              770.000000             11.000000        2.00000   
50%             1539.000000             33.900000        4.00000   
75%             1924.000000             83.000000        8.00000   
max            55000.000000         999479.419319      336.00000   

        fr_ma_rech90  sumamnt_ma_rech90  medianamnt_ma_rech90  \
count  209593.000000      209593.000000         209593.000000   
mean        7.716780       12396.218352           1864.595821   
std        12.590251       16857.793882           2081.680664   
min         0.000000           0.000000              0.000000   
25%         0.000000        2317.000000            773.000000   
50%         2.000000        7226.000000           1539.000000   
75%         8.000000       16000.000000           1924.000000   
max        88.000000      953036.000000          55000.000000   

       medianmarechprebal90  cnt_da_rech30   fr_da_rech30  cnt_da_rech90  \
count         209593.000000  209593.000000  209593.000000  209593.000000   
mean              92.025541     262.578110    3749.494447       0.041495   
std              369.215658    4183.897978   53885.414979       0.397556   
min             -200.000000       0.000000       0.000000       0.000000   
25%               14.600000       0.000000       0.000000       0.000000   
50%               36.000000       0.000000       0.000000       0.000000   
75%               79.310000       0.000000       0.000000       0.000000   
max            41456.500000   99914.441420  999809.240107      38.000000   

        fr_da_rech90    cnt_loans30   amnt_loans30  maxamnt_loans30  \
count  209593.000000  209593.000000  209593.000000    20959

from the above discription we see that almost our data is imbalanced, we have more counts of 1 than 0 in or target variable.
Almost every independent variable is highly skewed to right and has outliers. 
aon shows negative value at minimum as age (in terms of days) cannot be negative, daily_decr30 and daily_decr90 also shows genative value at minimum that too needs to be corrected.
last_rech_date_ma and last_rech_date_da - minimum value also here should be corrected.

I will 1st convert aon, daily_decr30,  daily_decr90, last_rech_date_ma and last_rech_date_da into absolute values to remove the negative value.

I will replace the values which are more than (Q3 + 1.5(IQR)) with Q3 + 1.5(IQR) and in some comlumns where  values are less than  (Q1-1.5(IQR)) with Q1-1.5(IQR)

In [7]:
#cleaning aon
data['aon']= abs(data['aon'])
data['aon'] = np.where(data['aon']>data['aon'].quantile(0.75) + 1.5*(data['aon'].quantile(0.75)-data['aon'].quantile(0.25)),data['aon'].quantile(0.75) + 1.5*(data['aon'].quantile(0.75)-data['aon'].quantile(0.25)),data['aon'])
data['aon'].describe()

count    209593.000000
mean        668.871098
std         513.525480
min           1.000000
25%         246.000000
50%         527.000000
75%         982.000000
max        2086.000000
Name: aon, dtype: float64

In [8]:
#cleaning daily_decr30 and daily_decr90
data['daily_decr30']= abs(data['daily_decr30'])
data['daily_decr90']= abs(data['daily_decr90'])

data['daily_decr30'] = np.where(data['daily_decr30']>data['daily_decr30'].quantile(0.75) + 1.5*(data['daily_decr30'].quantile(0.75)-data['daily_decr30'].quantile(0.25)),data['daily_decr30'].quantile(0.75) + 1.5*(data['daily_decr30'].quantile(0.75)-data['daily_decr30'].quantile(0.25)),data['daily_decr30'])
data['daily_decr90'] = np.where(data['daily_decr90']>data['daily_decr90'].quantile(0.75) + 1.5*(data['daily_decr90'].quantile(0.75)-data['daily_decr90'].quantile(0.25)),data['daily_decr90'].quantile(0.75) + 1.5*(data['daily_decr90'].quantile(0.75)-data['daily_decr90'].quantile(0.25)),data['daily_decr90'])
print(data['daily_decr30'].describe())
print('*****')
print(data['daily_decr90'].describe())

count    209593.000000
mean       4468.132828
std        5859.376128
min           0.000000
25%          42.453000
50%        1469.175667
75%        7244.000000
max       18046.320500
Name: daily_decr30, dtype: float64
*****
count    209593.000000
mean       4855.290617
std        6438.197071
min           0.000000
25%          42.693333
50%        1500.000000
75%        7802.790000
max       19442.935000
Name: daily_decr90, dtype: float64


In [9]:
#rental30 and rental90
data['rental30'] = np.where(data['rental30']<data['rental30'].quantile(0.25) - 1.5*(data['rental30'].quantile(0.75)-data['rental30'].quantile(0.25)),data['rental30'].quantile(0.25) - 1.5*(data['rental30'].quantile(0.75)-data['rental30'].quantile(0.25)),data['rental30'])
data['rental90'] = np.where(data['rental90']<data['rental90'].quantile(0.25) - 1.5*(data['rental90'].quantile(0.75)-data['rental90'].quantile(0.25)),data['rental90'].quantile(0.25) - 1.5*(data['rental90'].quantile(0.75)-data['rental90'].quantile(0.25)),data['rental90'])
data['rental30'] = np.where(data['rental30']>data['rental30'].quantile(0.75) + 1.5*(data['rental30'].quantile(0.75)-data['rental30'].quantile(0.25)),data['rental30'].quantile(0.75) + 1.5*(data['rental30'].quantile(0.75)-data['rental30'].quantile(0.25)),data['rental30'])
data['rental90'] = np.where(data['rental90']>data['rental90'].quantile(0.75) + 1.5*(data['rental90'].quantile(0.75)-data['rental90'].quantile(0.25)),data['rental90'].quantile(0.75) + 1.5*(data['rental90'].quantile(0.75)-data['rental90'].quantile(0.25)),data['rental90'])
print(data['rental30'].describe())
print('*******')
print(data['rental90'].describe())

count    209593.000000
mean       2197.069184
std        2552.708718
min       -4334.360000
25%         280.420000
50%        1083.570000
75%        3356.940000
max        7971.720000
Name: rental30, dtype: float64
*******
count    209593.000000
mean       2769.147968
std        3251.352605
min       -5552.035000
25%         300.260000
50%        1334.000000
75%        4201.790000
max       10054.085000
Name: rental90, dtype: float64


In [10]:
#last_rech_date_ma and last_rech_date_da

data['last_rech_date_da'] = abs(data['last_rech_date_da'])
data['last_rech_date_ma'] = abs(data['last_rech_date_ma'])
data['last_rech_date_ma'] = np.where(data['last_rech_date_ma']>data['last_rech_date_ma'].quantile(0.75) + 1.5*(data['last_rech_date_ma'].quantile(0.75)-data['last_rech_date_ma'].quantile(0.25)),data['last_rech_date_ma'].quantile(0.75) + 1.5*(data['last_rech_date_ma'].quantile(0.75)-data['last_rech_date_ma'].quantile(0.25)),data['last_rech_date_ma'])
data['last_rech_date_da'] = np.where(data['last_rech_date_da']>data['last_rech_date_da'].quantile(0.75) + 1.5*(data['last_rech_date_da'].quantile(0.75)-data['last_rech_date_da'].quantile(0.25)),data['last_rech_date_da'].quantile(0.75) + 1.5*(data['last_rech_date_da'].quantile(0.75)-data['last_rech_date_da'].quantile(0.25)),data['last_rech_date_da'])
print(data['last_rech_date_ma'].describe())
print('**********')
print(data['last_rech_date_da'].describe())

count    209593.000000
mean          4.977103
std           5.037840
min           0.000000
25%           1.000000
50%           3.000000
75%           7.000000
max          16.000000
Name: last_rech_date_ma, dtype: float64
**********
count    209593.0
mean          0.0
std           0.0
min           0.0
25%           0.0
50%           0.0
75%           0.0
max           0.0
Name: last_rech_date_da, dtype: float64


In [11]:
#last_rech_amt_ma
data['last_rech_amt_ma'] = np.where(data['last_rech_amt_ma']>data['last_rech_amt_ma'].quantile(0.75) + 1.5*(data['last_rech_amt_ma'].quantile(0.75)-data['last_rech_amt_ma'].quantile(0.25)),data['last_rech_amt_ma'].quantile(0.75) + 1.5*(data['last_rech_amt_ma'].quantile(0.75)-data['last_rech_amt_ma'].quantile(0.25)),data['last_rech_amt_ma'])
data['last_rech_amt_ma'].describe()

count    209593.000000
mean       1719.109131
std        1345.846661
min           0.000000
25%         770.000000
50%        1539.000000
75%        2309.000000
max        4617.500000
Name: last_rech_amt_ma, dtype: float64

In [12]:
#sumamnt_ma_rech30
data['sumamnt_ma_rech30'] = np.where(data['sumamnt_ma_rech30']>data['sumamnt_ma_rech30'].quantile(0.75) + 1.5*(data['sumamnt_ma_rech30'].quantile(0.75)-data['sumamnt_ma_rech30'].quantile(0.25)),data['sumamnt_ma_rech30'].quantile(0.75) + 1.5*(data['sumamnt_ma_rech30'].quantile(0.75)-data['sumamnt_ma_rech30'].quantile(0.25)),data['sumamnt_ma_rech30'])
data['sumamnt_ma_rech30'].describe()

count    209593.000000
mean       6861.071877
std        6612.799020
min           0.000000
25%        1540.000000
50%        4628.000000
75%       10010.000000
max       22715.000000
Name: sumamnt_ma_rech30, dtype: float64

In [13]:
#medianamnt_ma_rech30
data['medianamnt_ma_rech30'] = np.where(data['medianamnt_ma_rech30']>data['medianamnt_ma_rech30'].quantile(0.75) + 1.5*(data['medianamnt_ma_rech30'].quantile(0.75)-data['medianamnt_ma_rech30'].quantile(0.25)),data['medianamnt_ma_rech30'].quantile(0.75) + 1.5*(data['medianamnt_ma_rech30'].quantile(0.75)-data['medianamnt_ma_rech30'].quantile(0.25)),data['medianamnt_ma_rech30'])
data['medianamnt_ma_rech30'].describe()

count    209593.000000
mean       1475.941906
std        1079.392998
min           0.000000
25%         770.000000
50%        1539.000000
75%        1924.000000
max        3655.000000
Name: medianamnt_ma_rech30, dtype: float64

In [14]:
#medianmarechprebal30
data['medianmarechprebal30'] = np.where(data['medianmarechprebal30']<data['medianmarechprebal30'].quantile(0.25) - 1.5*(data['medianmarechprebal30'].quantile(0.75)-data['medianmarechprebal30'].quantile(0.25)),data['medianmarechprebal30'].quantile(0.25) - 1.5*(data['medianmarechprebal30'].quantile(0.75)-data['medianmarechprebal30'].quantile(0.25)),data['medianmarechprebal30'])
data['medianmarechprebal30'] = np.where(data['medianmarechprebal30']>data['medianmarechprebal30'].quantile(0.75) + 1.5*(data['medianmarechprebal30'].quantile(0.75)-data['medianmarechprebal30'].quantile(0.25)),data['medianmarechprebal30'].quantile(0.75) + 1.5*(data['medianmarechprebal30'].quantile(0.75)-data['medianmarechprebal30'].quantile(0.25)),data['medianmarechprebal30'])
data['medianmarechprebal30'].describe()

count    209593.000000
mean         58.219786
std          64.544132
min         -97.000000
25%          11.000000
50%          33.900000
75%          83.000000
max         191.000000
Name: medianmarechprebal30, dtype: float64

In [15]:
#medianmarechprebal90,medianamnt_ma_rech90 and sumamnt_ma_rech90
data['medianmarechprebal90'] = np.where(data['medianmarechprebal90']<data['medianmarechprebal90'].quantile(0.25) - 1.5*(data['medianmarechprebal90'].quantile(0.75)-data['medianmarechprebal90'].quantile(0.25)),data['medianmarechprebal90'].quantile(0.25) - 1.5*(data['medianmarechprebal90'].quantile(0.75)-data['medianmarechprebal90'].quantile(0.25)),data['medianmarechprebal90'])
data['medianmarechprebal90'] = np.where(data['medianmarechprebal90']>data['medianmarechprebal90'].quantile(0.75) + 1.5*(data['medianmarechprebal90'].quantile(0.75)-data['medianmarechprebal90'].quantile(0.25)),data['medianmarechprebal90'].quantile(0.75) + 1.5*(data['medianmarechprebal90'].quantile(0.75)-data['medianmarechprebal90'].quantile(0.25)),data['medianmarechprebal90'])
data['medianamnt_ma_rech90'] = np.where(data['medianamnt_ma_rech90']>data['medianamnt_ma_rech90'].quantile(0.75) + 1.5*(data['medianamnt_ma_rech90'].quantile(0.75)-data['medianamnt_ma_rech90'].quantile(0.25)),data['medianamnt_ma_rech90'].quantile(0.75) + 1.5*(data['medianamnt_ma_rech90'].quantile(0.75)-data['medianamnt_ma_rech90'].quantile(0.25)),data['medianamnt_ma_rech90'])
data['sumamnt_ma_rech90'] = np.where(data['sumamnt_ma_rech90']>data['sumamnt_ma_rech90'].quantile(0.75) + 1.5*(data['sumamnt_ma_rech90'].quantile(0.75)-data['sumamnt_ma_rech90'].quantile(0.25)),data['sumamnt_ma_rech90'].quantile(0.75) + 1.5*(data['sumamnt_ma_rech90'].quantile(0.75)-data['sumamnt_ma_rech90'].quantile(0.25)),data['sumamnt_ma_rech90'])
print(data['sumamnt_ma_rech90'].describe())
print('**********')
print(data['medianamnt_ma_rech90'].describe())
print('**********')
print(data['medianmarechprebal90'].describe())


count    209593.000000
mean      10878.720344
std       10789.012367
min           0.000000
25%        2317.000000
50%        7226.000000
75%       16000.000000
max       36524.500000
Name: sumamnt_ma_rech90, dtype: float64
**********
count    209593.000000
mean       1517.062192
std        1056.056717
min           0.000000
25%         773.000000
50%        1539.000000
75%        1924.000000
max        3650.500000
Name: medianamnt_ma_rech90, dtype: float64
**********
count    209593.000000
mean         55.956072
std          57.799928
min         -82.465000
25%          14.600000
50%          36.000000
75%          79.310000
max         176.375000
Name: medianmarechprebal90, dtype: float64


In [16]:
#payback30 and payback90
data['payback30'] = np.where(data['payback30']>data['payback30'].quantile(0.75) + 1.5*(data['payback30'].quantile(0.75)-data['payback30'].quantile(0.25)),data['payback30'].quantile(0.75) + 1.5*(data['payback30'].quantile(0.75)-data['payback30'].quantile(0.25)),data['payback30'])
data['payback90'] = np.where(data['payback90']>data['payback90'].quantile(0.75) + 1.5*(data['payback90'].quantile(0.75)-data['payback90'].quantile(0.25)),data['payback90'].quantile(0.75) + 1.5*(data['payback90'].quantile(0.75)-data['payback90'].quantile(0.25)),data['payback90'])
print(data['payback30'].describe())
print('******')
print(data['payback90'].describe())

count    209593.000000
mean          2.299777
std           3.030659
min           0.000000
25%           0.000000
50%           0.000000
75%           3.750000
max           9.375000
Name: payback30, dtype: float64
******
count    209593.000000
mean          2.894150
std           3.632511
min           0.000000
25%           0.000000
50%           1.666667
75%           4.500000
max          11.250000
Name: payback90, dtype: float64


In [17]:
#fr_ma_rech30
data['fr_ma_rech30'] = np.where(data['fr_ma_rech30']>38,38,data['fr_ma_rech30'])

In [18]:
#fr_ma_rech30 , cnt_da_rech30
data['cnt_da_rech30']= np.where(data['cnt_da_rech30']>16,16,data['cnt_da_rech30'])
data['fr_da_rech30']=np.where(data['fr_da_rech30']>21,21,data['fr_da_rech30'])

In [19]:
# cnt_loans90
data['cnt_loans90'] = np.where(data['cnt_loans90']>55,55,data['cnt_loans90'])


In [20]:
#maxamnt_loans30 and maxamnt_loans90
data['maxamnt_loans30'] = np.where(data['maxamnt_loans30']>12,np.median(data['maxamnt_loans30']),data['maxamnt_loans30'])
data['maxamnt_loans90'] = np.where(data['maxamnt_loans90']>12,np.median(data['maxamnt_loans90']),data['maxamnt_loans90'])

In [21]:
data.describe()

label            aon   daily_decr30   daily_decr90  \
count  209593.000000  209593.000000  209593.000000  209593.000000   
mean        0.875177     668.871098    4468.132828    4855.290617   
std         0.330519     513.525480    5859.376128    6438.197071   
min         0.000000       1.000000       0.000000       0.000000   
25%         1.000000     246.000000      42.453000      42.693333   
50%         1.000000     527.000000    1469.175667    1500.000000   
75%         1.000000     982.000000    7244.000000    7802.790000   
max         1.000000    2086.000000   18046.320500   19442.935000   

            rental30       rental90  last_rech_date_ma  last_rech_date_da  \
count  209593.000000  209593.000000      209593.000000           209593.0   
mean     2197.069184    2769.147968           4.977103                0.0   
std      2552.708718    3251.352605           5.037840                0.0   
min     -4334.360000   -5552.035000           0.000000                0.0   
25%       280.420000     300.260000           1.000000                0.0   
50%      1083.570000    1334.000000           3.000000                0.0   
75%      3356.940000    4201.790000           7.000000                0.0   
max      7971.720000   10054.085000          16.000000                0.0   

       last_rech_amt_ma  cnt_ma_rech30   fr_ma_rech30  sumamnt_ma_rech30  \
count     209593.000000  209593.000000  209593.000000      209593.000000   
mean        1719.109131       3.978057       4.065928        6861.071877   
std         1345.846661       4.256090       5.935577        6612.799020   
min            0.000000       0.000000       0.000000           0.000000   
25%          770.000000       1.000000       0.000000        1540.000000   
50%         1539.000000       3.000000       2.000000        4628.000000   
75%         2309.000000       5.000000       6.000000       10010.000000   
max         4617.500000     203.000000      38.000000       22715.000000   

       medianamnt_ma_rech30  medianmarechprebal30  cnt_ma_rech90  \
count         209593.000000         209593.000000   209593.00000   
mean            1475.941906             58.219786        6.31543   
std             1079.392998             64.544132        7.19347   
min                0.000000            -97.000000        0.00000   
25%              770.000000             11.000000        2.00000   
50%             1539.000000             33.900000        4.00000   
75%             1924.000000             83.000000        8.00000   
max             3655.000000            191.000000      336.00000   

        fr_ma_rech90  sumamnt_ma_rech90  medianamnt_ma_rech90  \
count  209593.000000      209593.000000         209593.000000   
mean        7.716780       10878.720344           1517.062192   
std        12.590251       10789.012367           1056.056717   
min         0.000000           0.000000              0.000000   
25%         0.000000        2317.000000            773.000000   
50%         2.000000        7226.000000           1539.000000   
75%         8.000000       16000.000000           1924.000000   
max        88.000000       36524.500000           3650.500000   

       medianmarechprebal90  cnt_da_rech30   fr_da_rech30  cnt_da_rech90  \
count         209593.000000  209593.000000  209593.000000  209593.000000   
mean              55.956072       0.102775       0.122805       0.041495   
std               57.799928       1.155806       1.543246       0.397556   
min              -82.465000       0.000000       0.000000       0.000000   
25%               14.600000       0.000000       0.000000       0.000000   
50%               36.000000       0.000000       0.000000       0.000000   
75%               79.310000       0.000000       0.000000       0.000000   
max              176.375000      16.000000      21.000000      38.000000   

        fr_da_rech90    cnt_loans30   amnt_loans30  maxamnt_loans30  \
count  209593.000000  209593.000000  209593.000000    20959

In [22]:
data.drop('last_rech_date_da',axis=1,inplace=True)

In [23]:
data.to_csv('data_cleaned_2.csv')